dbms/src/Functions/FunctionsLogical.h

```diff
@@ -240,7 +242,8 @@ struct AssociativeOperationImpl<Op, 1>
 
     inline UInt8 apply(size_t i) const
     {
-        return vec[i];
+        // Searching keyword "cast_bool_reason" in this file to get the reason for static_cast<bool>
+        return static_cast<bool>(vec[i]);
```

Comment:
maybe we can change the return type of `apply` to bool?

Reply:
> maybe we can change the return type of `apply` to bool?

okk